src/main.rs

```diff
@@ -46,10 +46,6 @@ pub fn mkdir(dir: &str) -> Result<()> {
     Ok(())
 }
 
-fn read_file(filename: &str) -> std::result::Result<String, String> {
-    read_to_string(filename).map_err(|e| format!("Failed to read {filename}: {e}"))
```

Comment:
The main reason for this function is the `map_err`. It extends the error message so say _which_ file could not be read. From what I've seen, `read_to_string()` does not include it.

Without it, since we propagate all errors to `main()` it becomes unclear which file failed.

Reply:
You're right, I didn't think of that. I moved the functions into a `util` module and applied some refactoring.